js/classes/BaseCalendar.js

```diff
@@ -0,0 +1,582 @@
+'use strict';
+
+const Store = require('electron-store');
+const { ipcRenderer } = require('electron');
+const {
+    hourMinToHourFormatted,
+    isNegative,
+    subtractTime,
+    sumTime,
+    validateTime
+} = require('../time-math.js');
+const {
+    formatDayId,
+    sendWaiverDay,
+    displayWaiverWindow
+} = require('../workday-waiver-aux.js');
+const { showDay, switchCalendarView } = require('../user-preferences.js');
+const { getDateStr, getMonthLength } = require('../date-aux.js');
+const { getMonthName } = require('../date-to-string-util.js');
+const { computeAllTimeBalanceUntilAsync } = require('../time-balance.js');
+const { generateKey } = require('../date-db-formatter.js');
+
+// Global values for calendar
+const flexibleStore = new Store({name: 'flexible-store'});
+const waivedWorkdays = new Store({name: 'waived-workdays'});
+
+// Holds the calendar information and manipulation functions
+class BaseCalendar
+{
+    /**
+     * @param {Object.<string, any>} preferences
+     */
+    constructor(preferences)
+    {
+        this._calendarDate = new Date();
+        this.loadInternalStore();
+        this.loadInternalWaiveStore();
+        this.updatePreferences(preferences);
+        this._initCalendar();
+    }
+
+    /**
+     * Initializes the calendar by generating the html code, binding JS events and then drawing according to DB.
+     */
+    _initCalendar()
+    {
+        throw Error('Please implement this.');
+    }
+
+    /**
+     * Returns a date object for which the all time balance will be calculated.
+     * If current month, returns the actual day. If not, first day of following month.
+     * @return {Date}
+     */
+    _getTargetDayForAllTimeBalance()
+    {
+        throw Error('Please implement this.');
```

Comment:
Maybe adding a dummy return here will please the deep code:

```suggestion
        throw Error('Please implement this.');
        return new Date();
```